ros2/nightly/nightly/Dockerfile

```diff
@@ -0,0 +1,80 @@
+# This is an auto generated Dockerfile for ros:nightly
+# generated from docker_images_ros2/nightly/create_ros_image.Dockerfile.em
+FROM ubuntu:bionic
+
+# setup timezone
+RUN echo 'Etc/UTC' > /etc/timezone && \
+    ln -s /usr/share/zoneinfo/Etc/UTC /etc/localtime && \
+    apt-get update && apt-get install -q -y tzdata && rm -rf /var/lib/apt/lists/*
+
+# install packages
+RUN apt-get update && apt-get install -q -y \
+    dirmngr \
+    gnupg2 \
+    lsb-release \
+    python3-pip \
+    && rm -rf /var/lib/apt/lists/*
+
+# setup ros2 keys
+RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 421C365BD9FF1F717815A3895523BAEEB01FA116
+
+# setup sources.list
+RUN echo "deb http://packages.ros.org/ros2/ubuntu `lsb_release -sc` main" > /etc/apt/sources.list.d/ros2-latest.list
+
+# install bootstrap tools
+RUN apt-get update && apt-get install --no-install-recommends -y \
+    python3-rosdep \
+    python3-rosinstall \
+    python3-vcstools \
+    wget \
+    && rm -rf /var/lib/apt/lists/*
+
+# setup environment
+ENV LANG C.UTF-8
+ENV LC_ALL C.UTF-8
+
+# bootstrap rosdep
+RUN rosdep init \
+    && rosdep update
+
+# install python packages
+RUN pip3 install -U \
+    argcomplete
+
+# install ros2 packages
+ENV ROS_DISTRO crystal
+RUN mkdir -p /opt/ros/$ROS_DISTRO
+ENV ROS2_BINARY_URL https://ci.ros2.org/view/packaging/job/packaging_linux/lastSuccessfulBuild/artifact/ws/ros2-package-linux-x86_64.tar.bz2
+RUN wget -q $ROS2_BINARY_URL -O - | \
+    tar -xj --strip-components=1 -C /opt/ros/$ROS_DISTRO
+RUN sed -i "s|^\(_colcon_prefix_chain_sh_COLCON_CURRENT_PREFIX\s*=\s*\).*$|\1/opt/ros/$ROS_DISTRO|" \
+      /opt/ros/$ROS_DISTRO/setup.sh && \
+    sed -i "s|^\(_colcon_prefix_sh_COLCON_CURRENT_PREFIX\s*=\s*\).*$|\1/opt/ros/$ROS_DISTRO|" \
+      /opt/ros/$ROS_DISTRO/local_setup.sh
+
+# ignore specified rosdep keys
+COPY core.yaml /etc/ros/rosdep/
```

Comment:
Is there a reason to inject these empty rosdeps rather than running with `--skip-keys` as the [instructions](https://index.ros.org/doc/ros2/Installation/Linux-Install-Binary/#id5) do? Since the list is maintained by hand a list of skipped keys would seem to me to be only as much work to maintain as the core.yaml.

Reply:
Answering my own question, the reason is so that "downstream" invocations of rosdep benefit from treating the same packages as resolved rather than having to propagate the skip list. :cool:.

Reply:
> I don't suspect this list changes frequently?

It changes when we introduce new prerequisites, which yeah, isn't often.